src/bittybuzz/src/BittyBuzzVm.cpp

```diff
@@ -0,0 +1,46 @@
+#include "bittybuzz/BittyBuzzVm.h"
+#include "bittybuzz/BittyBuzzSystem.h"
+#include <FreeRTOS.h>
+#include <bbzvm.h>
+#include <task.h>
+
+BittyBuzzVm::BittyBuzzVm(const IBittyBuzzBytecode& bytecode,
+                         const IBSP& bsp,
+                         const ILogger& logger,
+                         const FunctionRegister* functionRegisters,
+                         uint16_t lengthFunctionRegisters) :
+
+    m_bytecode(bytecode), m_bsp(bsp), m_logger(logger) {
+    // Init global variable
+    vm = &m_bbzVm;
+    BittyBuzzSystem::logger = &logger;
+
+    // Init vm
+    bbzvm_construct(m_bsp.getUUId());
+    bbzvm_set_error_receiver(BittyBuzzSystem::errorReceiver);
+    bbzvm_set_bcode(m_bytecode.getBytecodeFetchFunction(), m_bytecode.getBytecodeLength());
+    bbzringbuf_construct(&m_bbzPayloadBuff, m_bbzMsgBuff, 1, 11);
```

Comment:
This is the second time I see the number 11 somewhere. What is it? And could we maybe move it to a define?

Reply:
Raised it to 16, added a macro and changed to an std::array instead